react-d3-chart/src/components/Victory/Context/index.tsx

```diff
@@ -15,40 +16,30 @@ interface SelfProps {
   entireDomain: DomainPropObjectType;
   selectedDomain: DomainPropObjectType;
   handleDomainChange: (selectedDomain: DomainPropObjectType) => void;
-  colour?: d3.ScaleOrdinal<string, string>;
+  color?: d3.ScaleOrdinal<string, string>;
   tickCount?: number;
   data: LineProps[];
   maxPoints?: number;
 }
 
 export type ContextProps = SelfProps & Dimensions;
 
-const contextFilterData = (data: LineProps[], maxPoints: number) => {
-  if (data.length > 0) {
-    return data.map((line) => {
-      const k = Math.ceil(line.length / maxPoints);
-      return line.filter((_, i) => i % k === 0);
-    });
-  }
-  return [];
-};
-
 const Context: React.FC<ContextProps> = ({
   width,
   entireDomain,
   margin,
-  colour = DEFAULT_COLOUR,
+  color = DEFAULT_COLOR,
   height = CONTEXT_HEIGHT,
   maxPoints = 150,
   selectedDomain,
   handleDomainChange,
   tickCount,
   data,
 }) => {
-  const contextFilteredData = useMemo(
-    () => contextFilterData(data, maxPoints),
-    [data, maxPoints]
-  );
+  const contextFilteredData = useMemo(() => downSample(data, maxPoints), [
+    data,
```

Comment:
`downSample` take an array of coordinates and filters out coordinates from the array based on their `x position` % `skip rate` (Input: Array of original data points, Output: Array of filtered data points). Since the domain of the context never changes as apposed to the changing domain of the main graph, I use `useMemo` here to prevent unnecessary recalculations of the downSampled context data points.